src/guiguts/misc_dialogs.py

```diff
@@ -888,6 +889,140 @@ def insert_char(self, event: tk.Event) -> None:
         insert_in_focus_widget(char)
 
 
+class CommandPaletteDialog(ToplevelDialog):
+    """Command Palette Dialog."""
+
+    manual_page = "Help_Menu#Command_Palette"
+
+    def __init__(self) -> None:
+        """Initialize the command palette window."""
+        super().__init__("Command Palette")
+        self.commands = menubar_metadata().get_all_commands()
+        self.filtered_commands = self.commands
+
+        self.top_frame.grid_rowconfigure(0, weight=0)
+        self.top_frame.grid_rowconfigure(1, weight=1)
+
+        entry_frame = ttk.Frame(self.top_frame)
+        entry_frame.grid(row=0, column=0, sticky="NSEW", columnspan=2)
+        entry_frame.grid_columnconfigure(0, weight=1)
+        self.search_var = tk.StringVar()
+        self.search_var.trace_add("write", lambda _1, _2, _3: self.update_list())
+        self.entry = ttk.Entry(entry_frame, textvariable=self.search_var)
+        self.entry.grid(row=0, column=0, padx=5, pady=5, sticky="NSEW")
+        ToolTip(self.entry, "Type part of a command to filter the list")
+        ttk.Button(
+            entry_frame,
+            text="Run",
+            command=lambda: self.execute_command(tk.Event()),
+            takefocus=False,
+        ).grid(row=0, column=1, sticky="NSE")
+
+        columns = ("Command", "Shortcut", "Menu")
+        widths = (150, 120, 100)
+        self.list = ttk.Treeview(
+            self.top_frame,
+            columns=columns,
+            show="headings",
+            selectmode=tk.BROWSE,
+            height=20,
+        )
+        for col, column in enumerate(columns):
+            self.list.heading(f"#{col + 1}", text=column)
+            self.list.column(
+                f"#{col + 1}",
+                minwidth=10,
+                width=widths[col],
+                stretch=(col == 0),
+                anchor=tk.W if col == 0 else tk.CENTER,
+            )
+        self.list.grid(row=1, column=0, padx=5, sticky="NSEW")
+        self.scrollbar = ttk.Scrollbar(
+            self.top_frame, orient="vertical", command=self.list.yview
+        )
+        self.scrollbar.grid(row=1, column=1, sticky="NS")
+        self.list.config(yscrollcommand=self.scrollbar.set)
+
+        # Bind events for list and entry
+        self.list.bind("<Return>", self.execute_command)
+        self.list.bind("<Double-Button-1>", self.execute_command)
```

Comment:
Looking at some other implementations of palettes I'm familiar with (VScode, macOS Spotlight, Alfred on macOS), they all share the common trait that clicking once chooses, not double-click. What do you think of just having click select/execute instead of double-click? If not, is it because you want to reserve click for something in the future?

Reply:
> Looking at some other implementations of palettes I'm familiar with (VScode, macOS Spotlight, Alfred on macOS), they all share the common trait that clicking once chooses, not double-click. What do you think of just having click select/execute instead of double-click? If not, is it because you want to reserve click for something in the future?

I thought that clicking once to choose was pretty standard across systems, and double-clicking to execute. Or am I missing something?

Reply:
Oh ... clicking once on a button executes, though, and that's standard across platforms, too.

Reply:
In this case, it's a list of things for you to execute, so in my mind, choose == execute. Unless there is a future reason to select one for another purpose that @windymilla has in mind and thus it's kept in reserve for that use.

Reply:
One purpose I mentioned elsewhere (or intended to) is to add an "edit" button that allows you to edit/delete/add the shortcut for an entry.

I suspect it will be mostly used by people who are navigating it via the keyboard, so Return/Enter will execute. Or single click the entry then single click the button. Or double click the entry. Can we see if double click is too annoying, because I'd prefer to use double for execute if it's not.